pyvista/plotting/colors.py

```diff
@@ -761,9 +761,9 @@ def __init__(
                 # From vtkColor3ub instance (can be unpacked as rgb tuple)
                 self._from_rgba(color)
             else:
-                raise ValueError(f"Unsupported color type: {type(color)}")
+                raise TypeError(f"Unsupported color type: {type(color)}")
             self._name = color_names.get(self.hex_rgb, None)
-        except ValueError as e:
+        except TypeError as e:
```

Comment:
```suggestion
        except (ValueError, TypeError) as e:
```

The `_from_rgba` and `_from_str` methods can still raise a `ValueError`. I haven't checked the other changes here, but be careful there as well when changing the exception type.